### PR TITLE
Add engineStreamingGetUsedMemory

### PR DESCRIPTION
**Pull request by Pirulax**
Function for retrieving the amount of memory (in bytes) used by the GTA streamer.

### CODE REVIEW

**Comment by Pirulax** [comment]
Maybe `engineStreamingGetMemoryInUse` is a more fitting? Or `engineStreamingGetMemoryUsed`?

**Comment by AlexTMjugador** [comment]
I'd suggest `engineGetStreamingMemoryUsed`, because most engine functions follow the naming convention of `engine` + verb + object, so it's arguably more consistent. But your choice is as good as mine :smile:

**Comment by Pirulax** [comment]
It is more consistent, but I want to add more streaming functions in the feature, so I thought that placing `streaming` after `engine` might make those functions more consistent.

File: Client/sdk/game/CStreaming.h [hunk 1]
```diff
@@ -48,4 +48,5 @@ class CStreaming
     virtual void RequestSpecialModel(DWORD model, const char* szTexture, DWORD channel) = 0;
     virtual CStreamingInfo* GetStreamingInfoFromModelId(uint32 id) = 0;
     virtual void ReinitStreaming() = 0;
+    virtual std::uint32_t GetMemoryUsed() = 0;
```
Comment (botder) [inline Client/sdk/game/CStreaming.h]:
```suggestion
    virtual std::uint32_t GetMemoryUsed() const = 0;
```

File: Client/game_sa/CStreamingSA.cpp [hunk 2]
```diff
@@ -152,3 +152,8 @@ void CStreamingSA::ReinitStreaming()
 
     reinitStreaming();
 }
+
+std::uint32_t CStreamingSA::GetMemoryUsed()
+{
+    return *(std::uint32_t*)0x8E4CB4;
```
Comment (botder) [inline Client/game_sa/CStreamingSA.cpp]:
```suggestion
    return *reinterpret_cast<std::uint32_t*>(0x8E4CB4);
```

File: Client/game_sa/CStreamingSA.cpp [hunk 3]
```diff
@@ -152,3 +152,8 @@ void CStreamingSA::ReinitStreaming()
 
     reinitStreaming();
 }
+
+std::uint32_t CStreamingSA::GetMemoryUsed()
```
Comment (botder) [inline Client/game_sa/CStreamingSA.cpp]:
```suggestion
std::uint32_t CStreamingSA::GetMemoryUsed() const
```

File: Client/game_sa/CStreamingSA.h [hunk 4]
```diff
@@ -29,6 +29,7 @@ class CStreamingSA : public CStreaming
     void RequestSpecialModel(DWORD model, const char* szTexture, DWORD channel);
     void ReinitStreaming();
     CStreamingInfo* GetStreamingInfoFromModelId(uint32 id);
+    std::uint32_t GetMemoryUsed() override;
```
Comment (botder) [inline Client/game_sa/CStreamingSA.h]:
```suggestion
    std::uint32_t GetMemoryUsed() const override;
```

**Review by botder — CHANGES_REQUESTED** [review]

**Review by botder — APPROVED** [review]

**Pull request merged**